cluster/addons/metrics-server/resource-reader.yaml

```diff
@@ -16,26 +16,56 @@ rules:
   - get
   - list
   - watch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: system:metrics-server
+  labels:
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:metrics-server
+subjects:
+- kind: ServiceAccount
+  name: metrics-server
+  namespace: kube-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: system:metrics-server
+  namespace: kube-system
+  labels:
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+rules:
 - apiGroups:
-  - "extensions"
+  - extensions
+  - apps
   resources:
   - deployments
   verbs:
   - get
   - list
   - update
   - watch
+  resourceNames:
+  - metrics-server-v0.3.3
 ---
 apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRoleBinding
+kind: RoleBinding
 metadata:
   name: system:metrics-server
```

Comment:
I think we should rename this role and binding to something like `metrics-server-workaround`.

The name of this policy seems to [upset CI](https://buildkite.com/shopify/shopify-rbac-policies-policy-templates-validator/builds/929#50211ab8-98ea-4760-a152-a41c188913fd/63-1372)
```
system:metrics-server is invalid. Name must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character.
```

Even if that weren't the case, I think it's better for us to deploy this policy with a name that's **different** from the name that k8s uses. This will make our deployment and rollback of this change much easier.

If we use the same name:
- Deploying this will be complicated because we'll have to put the rbacbuddy annotation on each policy (otherwise rbacbuddy will refuse to modify the existing policy.)
- Rolling this back will be complicated because:
  - we'll have to remove the annotation on the policy *before* doing the cluster upgrade (otherwise rbacbuddy will revert the change from the k8s upgrade.)
  - we'll have to manually remove the annotation from each of the existing policies before removing this from the repo, otherwise rbacbuddy will remove the policy.